src/forward/arweave-fetcher.go

```diff
@@ -0,0 +1,224 @@
+						return
+					}
+					if len(interactions) == 0 {
+						return
+					}
+
+					// Update last_sort_key for each interaction in the database
+					// As a optimization lastSortKeys are cached in memory
+					lastSortKeys, err = self.updateLastSortKey(tx, interactions, height, lastSortKeys)
+					if err != nil {
+						return
+					}
+
+					// Update sync height
+					return self.updateSyncedHeight(tx, height)
+				})
+			if err != nil {
+				self.Log.WithError(err).Error("Failed to fetch interactions from DB")
+				return
+			}
+
+			if len(interactions) == 0 && offset != 0 {
+				// Edge case: num of interactions is a multiple of batch size
+				payload := &Payload{First: false, Last: true, Interaction: nil}
+				self.Output <- payload
+			} else {
+				isLastBatch := len(interactions) < self.Config.Forwarder.FetcherBatchSize
+				for i, interaction := range interactions {
+					payload := &Payload{
+						First:       isFirstBatch && i == 0,
+						Last:        isLastBatch && i == len(interactions)-1,
+						Interaction: interaction,
+					}
+
+					// NOTE: Quit only when the whole batch is processed
+					// That's why we're not waiting for closing of this task
+					self.Output <- payload
+				}
+			}
+
+			isFirstBatch = false
+		}
+	}
+	return
+}
+
+func (self *ArweaveFetcher) updateSyncedHeight(tx *gorm.DB, height uint64) (err error) {
+	var state model.State
+	err = tx.WithContext(self.Ctx).
+		Where("name = ?", model.SyncedComponentForwarder).
+		First(&state).
+		Error
+	if err != nil {
+		self.Log.WithError(err).Error("Failed to get state")
+		return err
+	}
+
+	if state.FinishedBlockHeight < height {
+		state.FinishedBlockHeight = height
+
+		err = tx.Model(&model.State{
+			Name: model.SyncedComponentForwarder,
+		}).
+			Updates(model.State{
+				FinishedBlockHeight: height,
+			}).
+			Error
+		if err != nil {
+			self.Log.WithError(err).Error("Failed to update stmonitorate after last block")
+			self.monitor.GetReport().Contractor.Errors.DbLastTransactionBlockHeight.Inc()
+			return err
+		}
+	}
+	return
+}
+
+func (self *ArweaveFetcher) updateLastSortKey(tx *gorm.DB, interactions []*model.Interaction, height uint64, lastSortKeys map[string]string) (out map[string]string, err error) {
+	// Get contract ids of fetched interactions
+	// Omit those that are already in the lastSortKeys map
+	newContractIds := self.getNewContractIds(interactions, lastSortKeys)
+
+	// Get last sort key for each contract
+	newLastSortKeys, err := self.getLastSortKeys(tx, newContractIds, height)
+	if err != nil {
+		return
+	}
+
+	// Merge new LSK into the existing map
+	out = tool.AppendMap(lastSortKeys, newLastSortKeys)
+
+	// Fill in last sort key for each interaction
+	for _, interaction := range interactions {
+		interaction.LastSortKey = out[interaction.ContractId]
+		out[interaction.ContractId] = interaction.SortKey
+	}
+
+	// Update last sort key for each contract
+	for _, interaction := range interactions {
+		err = tx.Model(interaction).
```

Comment:
is it possible to make a batch update here? instead of updating each row independently?